No longer possible to configure methods for the mock a class that does not exist

Since 3.2.4, using `getMock` on a class that does not exist no longer allows specifying methods that should be recognized. Calling a method on such a mock results in an undefined index warning.

This can be reproduced thanks to @nubs by following the steps described here : https://gist.github.com/nubs/ddb081e1d1b09541d7f7a1d93d009533

Classes that do not exist cannot be doubled using the new `createMock()` API:
```php
<?php
class Test extends PHPUnit\Framework\TestCase
{
    public function testClassThatDoesNotExistCanBeDoubledUsingMockBuilder()
    {
        $o = $this->createMock('DoesNotExist');

        $this->assertInstanceOf('DoesNotExist', $o);
    }
}
```
```
$ phpunit Test
PHPUnit 5.5.2-9-g88f407a by Sebastian Bergmann and contributors.

W                                                                   1 / 1 (100%)

Time: 19 ms, Memory: 4.00MB

There was 1 warning:

1) Test::testClassThatDoesNotExistCanBeDoubledUsingMockBuilder
Cannot stub or mock class or interface "DoesNotExist" which does not exist

WARNINGS!
Tests: 1, Assertions: 0, Warnings: 1.
```

Classes that do not exist can still be doubled using the Mock Builder API:
```php
<?php
class Test extends PHPUnit\Framework\TestCase
{
    public function testClassThatDoesNotExistCanBeDoubledUsingMockBuilder()
    {
        $o = $this->getMockBuilder('DoesNotExist')->getMock();

        $this->assertInstanceOf('DoesNotExist', $o);
    }
}
```
```
$ phpunit Test
PHPUnit 5.5.2-9-g88f407a by Sebastian Bergmann and contributors.

.                                                                   1 / 1 (100%)

Time: 21 ms, Memory: 4.00MB

OK (1 test, 1 assertion)
```
What is currently broken, and what I think this ticket is about, is configuring methods for test doubles of classes that do not exist:
```php
<?php
class Test extends PHPUnit\Framework\TestCase
{
    public function testClassThatDoesNotExistCanBeDoubledUsingMockBuilder()
    {
        $o = $this->getMockBuilder('DoesNotExist')
                  ->setMethods(['someMethod'])
                  ->getMock();
    }
}
```
```
PHPUnit 5.5.2-9-g88f407a by Sebastian Bergmann and contributors.

E                                                                   1 / 1 (100%)

Time: 19 ms, Memory: 4.00MB

There was 1 error:

1) Test::testClassThatDoesNotExistCanBeDoubledUsingMockBuilder
Undefined index: someMethod

/home/sb/Test.php:8

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.
```
This is a bug that I will fix ASAP.

--- a/src/Framework/MockObject/Generator.php
+++ b/src/Framework/MockObject/Generator.php
@@ -694,6 +694,7 @@ class PHPUnit_Framework_MockObject_Generator
         $cloneTemplate        = '';
         $isClass              = false;
         $isInterface          = false;
+        $isMultipleInterfaces = false;
 
         if (is_array($type)) {
             foreach ($type as $_type) {
@@ -705,6 +706,8 @@ class PHPUnit_Framework_MockObject_Generator
                         )
                     );
                 }
+
+                $isMultipleInterfaces = true;
 
                 $additionalInterfaces[] = $_type;
                 $typeClass              = new ReflectionClass($this->generateClassName(
@@ -833,7 +836,7 @@ class PHPUnit_Framework_MockObject_Generator
                     );
                 }
             }
-        } else {
+        } elseif ($isMultipleInterfaces) {
             foreach ($methods as $methodName) {
                 if ($this->canMockMethod($methodReflections[$methodName])) {
                     $mockedMethods .= $this->generateMockedMethodDefinitionFromExisting(

--- a/tests/GeneratorTest.php
+++ b/tests/GeneratorTest.php
@@ -198,7 +198,7 @@ class Framework_MockObject_GeneratorTest extends PHPUnit_Framework_TestCase
      */
     public function testCanConfigureMethodsForDoubleOfNonExistantClass()
     {
-        $className = md5(microtime());
+        $className = 'X' . md5(microtime());
 
         $mock = $this->generator->getMock($className, ['someMethod']);
 

